packages/gatsby-cli/src/init-starter.js

```diff
@@ -27,15 +28,56 @@ const shouldUseYarn = () => {
   }
 }
 
+const checkYarnVersion = () => {
+  let hasMinYarnPnp = false
+  let yarnVersion = null
+  try {
+    yarnVersion = execSync(`yarnpkg --version`)
+      .toString()
+      .trim()
+    hasMinYarnPnp = semver.gte(yarnVersion, `1.12.0`)
```

Comment:
will have to update this after https://github.com/yarnpkg/yarn/pull/6682 comes out (since Gatsby has a dependency, `relay-compiler` which is affected by this bug)